### PR TITLE
To Do list: add & remove

### LINKED ISSUE #2: [p2p] Improvement code
<h2>Hello @JoffreyNK 😀</h2>
Good work on your code
<h2>To highlight🎯</h2>
<ul>
<li>✔️ No linter errors</li>
<li>✔️ organized code </li>
<li>✔️ Nice design</li>
</ul>

<h2>Suggestions✍️</h2> 
✏In order to improve your code we suggest you add a placeholder for the accessibility of the user in the input;

https://github.com/JoffreyNK/To-Do-List/blob/e05053422698181d667884a33bb8e69feb16ee9c/src/index.html#L12

✏ You could kindly add an id to identify the correct checkbox checked to change the state to true.

https://github.com/JoffreyNK/To-Do-List/blob/e05053422698181d667884a33bb8e69feb16ee9c/src/modules/managedata.js#L74-L80

### PR DESCRIPTION
**Pull request by joffreynk**
# To Do list: add & remove

- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
 - Implemented a function for editing task descriptions.
 - Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
 - All changes to the To Do List are saved in local storage

### CODE REVIEW

File: src/modules/managedata.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  editTodo(index, description) {
+    this.todos = this.todos.map((singleTodo) => {
+      if (singleTodo.index === index) {
+        singleTodo.description = description;
+      }
+      return singleTodo;
+    });
+  }
```
Comment (youmari) [inline src/modules/managedata.js:36]: - [ ] please make sure when the task is edited is saved at the moment when I edit a task is not saved, please try to fix that

File: src/index.js [hunk 2]
```diff
@@ -1,64 +1,83 @@
...
+    localStorage.setItem('todo', JSON.stringify(todo.getTodos()));
+    populateData();
+  }
+
+  if (event.target.classList.contains('clear-button')) {
+    todo.clearCompleted();
+    localStorage.setItem('todo', JSON.stringify(todo.getTodos()));
+    populateData();
+  }
+});
```
Comment (youmari) [inline src/index.js:66]: - [ ] When removing a task you should update the indexes of all the remaining tasks as required please make sure to fix that 
image
image

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @JoffreyNK , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great title and short summary 
✔️ Good commit messages
✔️ No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joffreynk** [comment]
removing is working fine but the problem was editing and I have fixed it

**Review by mikethreels — APPROVED** [review]
Hi Joffrey, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**